wgpu-core/src/instance.rs

```diff
@@ -262,25 +267,49 @@ impl<I: Clone> AdapterInputs<'_, I> {
     }
 }
 
+struct Adapters<F: IdentityFilter<AdapterId>> {
```

Comment:
I made this struct to return data from `internal_enumerate_adapters`, I could also return a log tuple, if that's desired.